platform/ios/src/MGLMapView.mm

```diff
@@ -3824,10 +3824,12 @@ - (void)setShowsUserLocation:(BOOL)showsUserLocation
         if ([self.delegate respondsToSelector:@selector(mapView:viewForAnnotation:)])
         {
             userLocationAnnotationView = (MGLUserLocationAnnotationView *)[self.delegate mapView:self viewForAnnotation:self.userLocation];
-            if (userLocationAnnotationView)
+            if ( ! [userLocationAnnotationView isKindOfClass:MGLUserLocationAnnotationView.class])
             {
-                NSAssert([userLocationAnnotationView.class isSubclassOfClass:MGLUserLocationAnnotationView.class],
-                         @"User location annotation view must be a subclass of MGLUserLocationAnnotationView");
+                // Ignore the user-provided annotation if it is not a kind of
+                // MGLUserLocationAnnotationView (and use the default user
+                // location annotation instead).
```

Comment:
I think we should raise an exception in a future version, but I was worried about log spam. In testing, this message should only be shown when the user dot is initialized.

Here’s the log message I came up with:

> Ignoring incorrectly-typed user location annotation view. User location annotation view must be a kind of MGLUserLocationAnnotationView. This warning will become an error in a future version.